balloon.go

```diff
@@ -115,17 +115,26 @@ func (b *Instance) Mix(h hash.Hash, salt []byte, sCost, tCost uint64) {
 
 			for i := uint64(0); i < delta; i++ {
 				h.Reset()
-				binary.Write(h, binary.BigEndian, b.Cnt)
+				binary.Write(h, binary.LittleEndian, t)
+				binary.Write(h, binary.LittleEndian, m)
+				binary.Write(h, binary.LittleEndian, i)
+				idxBlock := h.Sum(nil)
+				h.Reset()
+				binary.Write(h, binary.LittleEndian, b.Cnt)
 				b.Cnt++
 				h.Write(salt)
-				binary.Write(h, binary.BigEndian, t)
-				binary.Write(h, binary.BigEndian, m)
-				binary.Write(h, binary.BigEndian, i)
-				otherInt.SetBytes(h.Sum(nil))
+				h.Write(idxBlock)
+				otherBytes := h.Sum(nil)
+
+				for left, right := 0, len(otherBytes)-1; left < right; left, right = left+1, right-1 {
+					otherBytes[left], otherBytes[right] = otherBytes[right], otherBytes[left]
+				}
```

Comment:
math/big only takes big-endian, I copied this reverse slice implementation from here:
https://github.com/golang/go/wiki/SliceTricks#reversing

Reply:
I was messing around with this project and found out that there exists a function in math/big that takes little-endian.

I've submitted a pull request on your fork a while ago, but sadly I think this original repo has gotten stale. If there's more interest, I might spend some time to optimize the implementation.

https://github.com/khonsulabs/balloon/pull/1

Reply:
Just saw your comment here, sorry for the delay!
@nogoegst has to look at this PR, I really don't feel comfortable reviewing unsafe Go code.

@darenliang personally I have no stake in this library, I was only trying to help multiple Balloon implementations be compatible with each other, specifically the [Rust library I contributed to](https://crates.io/crates/balloon-hash). So if you are more interested in using Balloon with Go and want to have common compatibility, feel free to contact me.